expression/builtin_time.go

```diff
@@ -1740,7 +1740,38 @@ type builtinMakeDateSig struct {
 
 // See https://dev.mysql.com/doc/refman/5.7/en/date-and-time-functions.html#function_makedate
 func (b *builtinMakeDateSig) eval(row []types.Datum) (d types.Datum, err error) {
-	return d, errFunctionNotExists.GenByArgs("MAKEDATE")
+	args, err := b.evalArgs(row)
+	if err != nil {
+		return d, errors.Trace(err)
+	}
+	if args[0].IsNull() || args[1].IsNull() {
+		return d, nil
+	}
+	year := args[0].GetInt64()
```

Comment:
Use args[0].ToInt64(). MySQL also accept other type of input:
mysql> select makedate("71", 1);
+-------------------+
| makedate("71", 1) |
+-------------------+
| 1971-01-01        |
+-------------------+
1 row in set (0.00 sec)